tests/org.eclipse.n4js.expectmatrix.tests/xpect-tests/matrix/Interface_x_ST_DS/02_as_return_value.n4js.xt

```diff
@@ -34,7 +34,7 @@ class C1 {
 	// return value must be nominal type XPECT errors --> "Structural type ~Object with { x: int; foo: {function():void} } is not a subtype of non-structural type I<int>." at "{x: 3, foo: this.f}"
 	public f1 () : I<int> { return {x: 3, foo: this.f}; };
 
-	// return value must be ~I, missing foo field XPECT errors --> "~Object with { x: ~int } is not a structural subtype of ~I<int>: missing method foo." at "{x: 3}"
+	// return value must be ~I, missing foo field XPECT errors --> "~Object with { x: union{int,~int} } is not a structural subtype of ~I<int>: missing method foo." at "{x: 3}"
```

Comment:
As discussed. The old code simplified this, but probably it was just an unintentional side effect.

It's true, we could simplify the composed type in such a situation, but since it is not critical and nothing is rendered invalid, I'd propose to not create a ticket for it right now but to do it if/when we run into it again.